geodiff/src/drivers/sqlitedriver.cpp

```diff
@@ -218,25 +234,37 @@ std::vector<std::string> SqliteDriver::listTables( bool useModified )
 
     tableNames.push_back( tableName );
   }
+  if ( rc != SQLITE_DONE )
+  {
+    std::string errMsg = sqliteErrorMessage( mDb->get(), "SqliteDriver::listTables" );
+    context()->logger().error( errMsg );
+  }
 
   // result is ordered by name
   return tableNames;
 }
 
-bool tableExists( std::shared_ptr<Sqlite3Db> db, const std::string &tableName, const std::string &dbName )
+bool tableExists( const Context *context, std::shared_ptr<Sqlite3Db> db, const std::string &tableName, const std::string &dbName )
 {
   Sqlite3Stmt stmtHasGeomColumnsInfo;
   stmtHasGeomColumnsInfo.prepare( db, "SELECT name FROM \"%w\".sqlite_master WHERE type='table' "
                                   "AND name='%q'", dbName.c_str(), tableName.c_str() );
-  return sqlite3_step( stmtHasGeomColumnsInfo.get() ) == SQLITE_ROW;
+  int rc = sqlite3_step( stmtHasGeomColumnsInfo.get() );
+  if ( rc != SQLITE_ROW )
+  {
+    std::string errMsg = sqliteErrorMessage( db->get(), "tableExists" );
+    context->logger().error( errMsg );
+  }
```

Comment:
you may regularly ask if table exists in the code, and that is not a error when the table does not exists. does this loog error even for this case?